pkg/deploy/strategy/rolling/rolling.go

```diff
@@ -121,6 +149,10 @@ func (s *RollingDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	params := config.Spec.Strategy.RollingParams
 	updateAcceptor := s.getUpdateAcceptor(time.Duration(*params.TimeoutSeconds)*time.Second, config.Spec.MinReadySeconds)
 
+	// Record all warnings
+	defer s.recordControllerWarnings(from)
```

Comment:
These will be executed twice on initial deployments. You should move them below the if clause you have just next to you.